### PR TITLE
MsSQL TRY_CONVERT

### PR DESCRIPTION
**Pull request by yoavcloud**
This PR adds support to parse the MsSQL TRY_CONVERT function, similar to CONVERT

### CODE REVIEW

File: src/parser/mod.rs [hunk 1]
```diff
@@ -1023,7 +1023,8 @@ impl<'a> Parser<'a> {
                     self.parse_time_functions(ObjectName(vec![w.to_ident()]))
                 }
                 Keyword::CASE => self.parse_case_expr(),
-                Keyword::CONVERT => self.parse_convert_expr(),
+                Keyword::CONVERT => self.parse_convert_expr(false),
+                Keyword::TRY_CONVERT if dialect_of!(self is MsSqlDialect) => self.parse_convert_expr(true),
```
Comment (iffyio) [inline src/parser/mod.rs:1027]: Can we have the guard as a dialect method e.g. `self.dialect.supports_try_convert()`? We can probably include support for the generic dialect as well

File: tests/sqlparser_mssql.rs [hunk 2]
```diff
@@ -495,6 +497,8 @@ fn parse_convert() {
         ParserError::ParserError("Expected: an expression, found: )".to_owned()),
         ms().parse_sql_statements(error_sql).unwrap_err()
     );
+
+    ms().verified_expr("TRY_CONVERT(VARCHAR(MAX), 'foo')");
```
Comment (iffyio) [inline tests/sqlparser_mssql.rs:501]: with the dialect method we can move this over to common.rs  as a generic test over the dialects that support the feature

**Review by iffyio — COMMENTED** [review]
@yoavcloud can we also add the flag to generic dialect? Other than that LGTM!

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Review by alamb — APPROVED** [review]
Thanks @yoavcloud  and @iffyio

**Comment by alamb** [comment]
Thanks agian @yoavcloud and @iffyio

**Pull request merged**